### PR TITLE
Add algo/localplan

### PR DESCRIPTION
**Pull request by bowei**
Adds algo/local_plan

Local plan computes a local plan of action for each pair of nodes in the (got, want) graph pair.

### CODE REVIEW

**Comment by briantkennedy** [comment]
This has some already added files listed as add, can you rebase?

**Comment by bowei** [comment]
Fixed -- thanks!

**Comment by bowei** [comment]
Ok, this should be ready now.

File: pkg/cloud/rgraph/algo/localplan/local_plan.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode"
+)
+
+// Do computes a plan local to each Node in the graph and puts the resulting
+// plan in the "want" Graph. It is required that got and want have the same set
+// of Nodes; Nodes that don't exist need to be marked as with
+// NodeStateDoesNotExist.
+func Do(got, want *rgraph.Graph) error {
```
Comment (briantkennedy) [inline pkg/cloud/rgraph/algo/localplan/local_plan.go:30]: The input/output var won't be obvious at the callsite when reading, can we call this `PopulateWant` or something that conveys we are updating the input and which input is getting modified?  Same idea for `Plan` where it's not quite obvious.

**Review by briantkennedy — APPROVED** [review]

**Pull request merged**